substring/main.c

```diff
@@ -0,0 +1,38 @@
+#include "findSubstring.h"
+#include "readTextFromFile.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int main()
+{
+    int testFailed = test();
+    if (testFailed != 0)
+    {
+        printf("TEST %d FAILED", testFailed);
+        return -1;
+    }
+
+    char* text = readTextFromFile("text.txt");
+    if (text == NULL)
+    {
+        return -1;
+    }
+
+    printf("Enter the pattern to find: ");
+    char pattern[100] = { 0 };
+    fgets(pattern, 100, stdin);
+    pattern[strlen(pattern) - 1] = '\0';
+
+    int index = KMPSearch(pattern, text);
```

Comment:
warning: implicit declaration of function ‘KMPSearch’; did you mean ‘bsearch’? [-Wimplicit-function-declaration]